src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriBuilder.java

```diff
@@ -406,16 +528,14 @@ public boolean addBundleTransportUpdateReq(
                 IceUdpTransportPacketExtension
                     .cloneTransportAndCandidates(transport, true));
 
-        // Note: if the request already contains a bundle with the same ID, the
-        // OLD one is kept.
-        boolean hasAnyChanges = request.addChannelBundle(channelBundleRequest);
-        if (!hasAnyChanges)
-        {
-            logger.warn("An channel-bundle update has been lost (an instance "
-                            + "with its ID already exists)");
-        }
+        request.addChannelBundle(channelBundleRequest);
 
-        return hasAnyChanges;
+        // Note that we don't actually check whether the addition of the bundle
+        // made any changes. We return true, because it is safe. It might lead
+        // us to send an unnecessary request, but it will not fail to send a
+        // request when we need it. We may need to add a check for modification
+        // of the content of the channel bundle extension.
+        return true;
```

Comment:
was there a reason to change the logic here to this?

Reply:
Yeah, because of the updated semantics of request.addChannelBundle(). It used to keep the old value, now it replaces the old value with the new value (which I think makes more sense).

So if we execute this twice, we can't easily check if the second call resulted in an update or not.

Reply:
ok, so `addChannelBundle` will replace an existing bundle if there was one?  maybe we should rename it to `setChannelBundle` ?

Reply:
What would you expect addChannelBundle to do in this case? Replacing is consistent with Map.put()

Reply:
i'd expect `addChannelBundle` to add this new bundle, without removing anything that was existing. `setChannelBundle` to me says: "set this as THE one and only channel bundle, replacing anything that was there".  put i think matches 'set' more closely than add.

Reply:
But there isn't only one channel bundle. In XML there's multiple child elements, and in java there's a map. I'd find a chain of "setChannelBundle" calls quite strange.

Reply:
ok, when you said "It used to keep the old value, now it replaces the old value with the new value" it sounded like whenever 'add' got called, it replaced the old one with the current one, meaning there could only be one bundle set.  did i misunderstand that?

Reply:
gotcha.